src/api.rs

```diff
@@ -674,6 +679,8 @@ pub(crate) struct ContextInner<T: Pixel> {
   ///  memory for the whole life of the encode.
   // TODO: Is this needed at all?
   keyframes: BTreeSet<u64>,
+  // TODO: Is this needed at all?
+  keyframes_forced: BTreeSet<u64>,
```

Comment:
Maybe related to the old code but... why is this a tree?

Reply:
HashSet seemed overkill.

Reply:
Not a simple vec/heap array/queue?

Reply:
A set makes sense in my opinion for the O(1) contains checks. We're doing contains checks much more frequently than inserts on both keyframes fields.